hq/app/aws/ec2/EC2.scala

```diff
@@ -124,24 +124,23 @@ object EC2 {
   }
 
   private[ec2] def addVpcName(flaggedSgs: List[SGOpenPortsDetail], vpcs: Map[String, Vpc]): List[SGOpenPortsDetail] = {
-    for {
-      sgs <- flaggedSgs
-      vpc <- vpcs.collectFirst { case (id, v) if id == sgs.vpcId => v }
-      vpcN = vpc.getTags.asScala.collectFirst { case tag if tag.getKey == "Name" => tag.getValue }
-      sgsUpdated = sgs.copy(vpcName = vpcN)
-    } yield sgsUpdated
+    def vpcName(vpc: Vpc) = vpc.getTags.asScala.collectFirst { case tag if tag.getKey == "Name" => tag.getValue }
+
+    flaggedSgs.map {
+      case s if s.vpcId.nonEmpty => s.copy(vpcName = vpcs.get(s.vpcId) flatMap vpcName)
```

Comment:
As discussed IRL... 😀 

If we are certain that the `.get(` here cannot fail if the `vpcId` doesn't match anything in the `vpcs`, then this looks okay to me 👍